### ISSUE TITLE
[BC] Add wrapper function in core/includes/drupal.inc for `theme_dblog_message()`

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#5763]
This function was removed early on in Backdrop 1.0.0 (see #539 and https://docs.backdropcms.org/change-records/theme_dblog_message-has-been-removed), but we never added a wrapper function for it (for backwards compatibility). The function was replaced by `dblog_format_message()`.

**Comment by bugfolder** [comment]
I tried testing this (in PHP 8.1) by enabling the Devel module and using "Execute PHP". Since this was a Drupal function originally, I used code from [the D7 `dblog_event()` function](https://api.drupal.org/api/drupal/modules%21dblog%21dblog.admin.inc/function/dblog_event/7.x) as the model:

```
require_once BACKDROP_ROOT . '/' . backdrop_get_path('module', 'dblog') . '/dblog.admin.inc';
$dblog = db_query('SELECT w.*, u.name, u.uid FROM {watchdog} w INNER JOIN {users} u ON w.uid = u.uid LIMIT 1', array())->fetchObject();
$output = theme_dblog_message(array('event' => $dblog));
dpm($output);
```

After patching, watchdog got the deprecation notice "The function eval() (devel.module(1184) : eval()'d code:3) called deprecated function theme_dblog_message(). It will be removed in the next major release of Backdrop.", as expected.

However, there were two more notices in watchdog:

* Warning: Undefined array key "link" in theme_dblog_message() (line 2589 of /mysite/core/includes/drupal.inc).
* Warning: Undefined variable $is_link in theme_dblog_message() (line 2593 of /mysite/core/includes/drupal.inc).

Seems like those shouldn't be generated, just the deprecation notice.

Also, one is not supposed to call `theme_XXX()` functions directly, instead one should call `theme('XXX', ...)`. So a more realistic test would be

```
require_once BACKDROP_ROOT . '/' . backdrop_get_path('module', 'dblog') . '/dblog.admin.inc';
$dblog = db_query('SELECT w.*, u.name, u.uid FROM {watchdog} w INNER JOIN {users} u ON w.uid = u.uid LIMIT 1', array())->fetchObject();
$output = theme('dblog_message', array('event' => $dblog));
dpm($output);
```

And this results in watchdog report "Theme hook "dblog_message" not found." If we're going to provide a `theme_dblog_message()` function, do we need to define the `dblog_message` object in a `hook_theme` implementation?

**Comment by argiepiano** [comment]
Yes, for this to work as a theme function (and avoid those other warning) we need a definition of the hook in hook_theme() somewhere. 

Typically:
```php
    'dblog_message' => array(
      'variables' => array('event' => NULL, 'link' => FALSE),
      'file' => 'dblog.admin.inc',
    ),
```

That will initialize the `$link` variable. In fact, this would bring back the whole theming function, which was removed earlier on. Why?

I wonder if we really need this. There is no instance in contrib of a call to this theme, and all its use has been expunged from core. It's one of those theming hooks that are internal and nobody really uses. I wish the OP had explained why this is needed. Was ia warning somewhere? A port issue?
